File: frontend/src/types/country.ts
export interface CountryBasic {
  iso_code: string;
  name: string;
  flag?: string;
  region?: string;
}

export interface CountryInfo {
  iso_code: string;
  name: string;
  official_name?: string;
  capital?: string[];
  population?: number;
  area?: number;
  region?: string;
  subregion?: string;
  languages?: Record<string, string>;
  currencies?: Record<string, { name: string; symbol: string }>;
  flag?: string;
  coat_of_arms?: string;
  map_url?: string;
  latlng?: number[];
  wikipedia_summary?: string;
  wikipedia_thumbnail?: string;
}

File: backend/app/schemas/country.py
from pydantic import BaseModel


class CountryBasic(BaseModel):
    name: str
    iso_code: str
    flag: str = ""
    region: str = ""


class CountryInfo(BaseModel):
    name: str
    official_name: str = ""
    iso_code: str
    capital: list[str] = []
    region: str = ""
    subregion: str = ""
    population: int = 0
    area: float = 0.0
    languages: dict[str, str] = {}
    currencies: dict[str, dict[str, str]] = {}
    flag: str = ""
    coat_of_arms: str = ""
    map_url: str = ""
    latlng: list[float] = []
    wikipedia_summary: str = ""
    wikipedia_thumbnail: str = ""
